borsh/src/ser/helpers.rs

```diff
@@ -0,0 +1,23 @@
+use crate::maybestd::{
+    io::{Result, Write},
+    vec::Vec,
+};
+use crate::BorshSerialize;
+
+/// Serialize an object into a vector of bytes.
+#[inline]
+pub fn serialize_to_vec<T>(value: &T) -> Result<Vec<u8>>
+where
+    T: BorshSerialize + ?Sized,
+{
+    value.try_to_vec()
+}
+
+/// Serializes an object directly into a `Writer`.
+#[inline]
+pub fn serialize_to_writer<T, W: Write>(value: &T, mut writer: W) -> Result<()>
```

Comment:
```suggestion
pub fn serialize_to_writer<T, W: Write>(mut writer: W, value: &T) -> Result<()>
```

both sered_json and bincode place the writer first, so let's be consistent with them. In general, I think it's good practice to place "context" arguments first

Reply:
swapped